core/src/banking_stage/consumer.rs

```diff
@@ -472,14 +472,23 @@ impl Consumer {
             ..
         } = execute_and_commit_transactions_output;
 
+        // The transaction scheduler may allocate costs for transactions that don't get executed
```

Comment:
This comment implies we care about execution rather than commitment, which is contrasted w/ the comments in the function itself.

I think what we really care about is whether or not the transaction was recorded, i.e. inserted in the block. We can get this indirectly through the commit transactions result.

wdyt of something like:
```rust
        // Costs of all transactions are added before processing.
        // To ensure accurate tracking of compute units, transactions that 
        // ultimately were not included in the block should have their
        // cost removed.
```

Reply:
works for me!